log/log.go

```diff
@@ -212,8 +211,7 @@ func (l Logger) log(lvl level, f F) {
 }
 
 func (l Logger) flush() {
-	b := &bytes.Buffer{}
-	encoder := json.NewEncoder(b)
+	encoder := json.NewEncoder(l.w)
```

Comment:
This branch is probably fast because the benchmark used has the writer been `io.Discard`   
However in real world logging situations, the writer is `os.Stdout`.  
The call below[ie, `encoder.Encode(v)` ] writes to the writer passed into `json.NewEncoder()`   
So if the writer is stdout, it will be written to for every loop.  
That sounds worse than the current main branch where the write to stdout happens only once.